sdk/python/packages/flet-core/src/flet_core/utils/datetime.py

```diff
@@ -0,0 +1,30 @@
+import re
+from datetime import datetime, date
+from typing import Optional, Union
+
+
+def has_valid_datetime_format(value: str) -> bool:
+    """
+    Checks if the given string has a valid datetime format.
+    Returns True if that's the case and False otherwise.
+    """
+    regex_format = (
+        r"^([+-]?\d{4,6})-?(\d\d)-?(\d\d)"
+        r"(?:[ T](\d\d)(?::?(\d\d)(?::?(\d\d)(?:[.,](\d+))?)?)?"
+        r"( ?[zZ]| ?([-+])(\d\d)(?::?(\d\d))?)?)?$"
+    )
+    return re.match(regex_format, value) is not None
+
+
+def datetime_to_string(value: Optional[Union[datetime, str]]) -> Optional[str]:
+    """
```

Comment:
Do we really want to allow date/time input as a string?
What date format does that Regex validate?

Reply:
> What date format does that Regex validate?

That regex comes from [Flutter's datetime parser](https://api.flutter.dev/flutter/dart-core/DateTime/parse.html), which we make use of:
https://github.com/flet-dev/flet/blob/46f3f4cee329b7abfd065918ebfd7e65e07a6fff/packages/flet/lib/src/models/control.dart#L105

> Do we really want to allow date/time input as a string?

Actually, no.

Reply:
OK, why do we need that regex then?

Reply:
Checkout the issue: https://github.com/flet-dev/flet/issues/4017

Reply:
Sure, but the issue with this is "consistency throughout". 
If we start being strict (by forcing/imposing a particular type of input) we will then have to do this everywhere. 
In enum-props for example, we do not impose the use of the respective enums, hence allowing the use of strings.

Do we want to be strict? :)

Reply:
For datetime I would never give you a correct string representation.
You are expecting some "ISO" format checked by that heavy regex. Looking at that regex I can't tell what would be correct case.
Also, solving datetime parsing shouldn't be on us. What if there is `mm/dd/yy` (US) or `dd/mm/yyyy` (EU) or `dd-mm-yy`, etc.?

enums are OK though.

Reply:
Datetime parsing depends on user locale and OS settings.
Let's just require only datetime object, no strings. Let's be strict.